src/lxilua.c

```diff
@@ -187,6 +192,11 @@ static int scpi(lua_State *L)
     int timeout = lua_tointeger(L, 3);
     char command_buffer[1000];
 
+    if(device < 0)
+    {
+        lua_pushlstring(L, -1);
```

Comment:
I think this should be lua_pushnumber(L, -1); the same way it returns in the end of this function.